gratipay/utils/sentry.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import sys
+import traceback
+
+from aspen import log
+from gratipay import wireup
+
+
+log_to_console = lambda exc_type, state: log(traceback.format_exc())
+
+
+class teller(object):
+    """This is a context manager to log to Sentry. You have to pass in an
+    ``Environment`` object with a ``sentry_dsn`` attribute.
+    """
+
+    def __init__(self, env, fallback=log_to_console):
+        try:
+            sys.stdout = sys.stderr  # work around aspen.log_dammit limitation; sigh
+            self.tell_sentry = wireup.make_sentry_teller(env, fallback)
+        finally:
+            sys.stdout = sys.__stdout__
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.tell_sentry(exc_type, {})
```

Comment:
Sentry accesses Python's global exception state directly during [`captureException`](https://github.com/gratipay/gratipay.com/blob/2072/gratipay/wireup.py#L163) (via `sys.exc_info`, presumably), so we don't have to pass it through these function calls.